piksi_tools/action_logger.py

```diff
@@ -0,0 +1,242 @@
+      not yet dispatchedm message received by device
+    """
+    msg = dispatch(msg)
+    if isinstance(msg, MsgTrackingState):
+      if self.debug:
+        print "currently tracking {0} sats".format(self.num_tracked_sats)
+      self.num_tracked_sats = 0
+      for channel, track_state in enumerate(msg.states):
+        prn = track_state.prn + 1
+        if track_state.state == 1:
+          self.num_tracked_sats += 1
+          self.prn_status_dict[prn] = channel
+          self.channel_status_dict[channel] = prn
+        else:
+          if self.prn_status_dict.get(prn):
+            del self.prn_status_dict[prn]
+          if self.channel_status_dict.get(channel):
+            del self.channel_status_dict[channel]
+
+  def drop_prns(self, prns):
+    """ drop prn array via sbp MsgMaskSatellite
+    Parameters
+    ----------
+    prns : int[]
+      list of prns to drop
+    """
+    if self.debug:
+      print "Dropping the following prns {0}".format(prns)
+    for prn in prns:
+      msg = MsgMaskSatellite(mask=2, prn=int(prn)-1)
+      self.handler.send_msg(msg)
+
+  def get_num_sats_to_drop(self):
+    """ return number of satellites to drop.
+    Should drop a random number of satellites above self.min_sats
+    If we haven't achieved min sats, it drops zero
+    """
+    max_to_drop = max(0, self.num_tracked_sats-self.min_sats)
+    # end points included
+    return random.randint(0, max_to_drop)
+
+  def drop_random_number_of_sats(self):
+    """ perform drop of satellites
+    """
+    num_drop = self.get_num_sats_to_drop()
+    if num_drop > 0:
+      prns_to_drop = random.sample(self.channel_status_dict.values(), num_drop)
+      if self.debug:
+        print ("satellite drop triggered: "
+                "will drop {0} out of {1} sats").format(num_drop,
+                                                          self.num_tracked_sats)
+      self.drop_prns(prns_to_drop)
+
+  def action(self):
+    """ overload of
+    """
+    self.drop_random_number_of_sats()
+
+def get_args():
+  """
+  Get and parse arguments.
+  """
+  import argparse
+  parser = sl.base_options()
+  parser.add_argument("-i", "--interval",
+                      default=[DEFAULT_POLL_INTERVAL], nargs=1,
+                      help="Number of seconds between satellite drop events.")
+  parser.add_argument("-m", "--minsats",
+                      default=[DEFAULT_MIN_SATS], nargs=1,
+                      help="Minimum number of satellites to retain during drop events.")
+  return parser.parse_args()
+
+def main():
+  """
+  Get configuration, get driver, get logger, and build handler and start it.
+  """
+  args = get_args()
+  port = args.port[0]
+  baud = args.baud[0]
+  timeout = args.timeout[0]
+  log_filename = args.log_filename[0]
+  append_log_filename = args.append_log_filename[0]
+  watchdog = args.watchdog[0]
+  tags = args.tags[0]
+  interval = int(args.interval[0])
+  minsats = int(args.minsats[0])
+
+  #initialize state machines:
```

Comment:
Orphaned comment that should be removed? Or some initialization that's not happening?